File: CompetitiveProgramming/LeetCode/problems/rust/1_two_sum/src/main.rs
use std::collections::HashMap;

fn main() {
    println!("{:?}",two_sum(vec![2,7,11,15], 9));
}


pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32>{
    let mut hm = HashMap::new();

    for (i, &num) in nums.iter().enumerate() {
        match hm.get(&num) {
            Some(&j) => return vec![j as i32, i as i32],
            None => {
                hm.insert(target - num, i);
            }
        }
    }

    unreachable!();
}

File: CompetitiveProgramming/LeetCode/problems/java/1_two_sum/src/Main.java
import java.util.HashMap;

public class Main {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
    }

    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length ; i++) {
            if (map.get(nums[i]) != null) {
                return new int[]{map.get(nums[i]), i};
            }
            map.putIfAbsent(target - nums[i], i);
        }

        return new int[]{};
    }
}